app/controllers/webauthn_credentials_controller.rb

```diff
@@ -53,4 +55,23 @@ def build_webauthn_credential
       )
     )
   end
+
+  def render_callback_redirect
+    if current_user.totp_disabled? && current_user.count_webauthn_credentials == 1
+      current_user.mfa_recovery_codes = Array.new(10).map { SecureRandom.hex(6) }
+      current_user.save!(validate: false)
+
+      session[:show_recovery_codes] = true
+      render json: { redirect_url: recovery_multifactor_auth_url }
+    else
+      render json: { redirect_url: edit_settings_url }
+    end
+  end
+
+  def set_recovery_codes_destroy
+    return unless current_user.count_webauthn_credentials == 1 && current_user.totp_disabled?
```

Comment:
Fixed in https://github.com/rubygems/rubygems.org/pull/3821/commits/29f31236e7bf68127a4335e5a438ef7bb56e1d9c

> Not sure why the tests didn't catch this 🤔

I think it might be because 2 of the setups called destroy https://github.com/rubygems/rubygems.org/pull/3821#discussion_r1209447678 (lol I swear we addressed that before since it was marked resolved). Once merged into one case, the recovery code test failed as expected.